data_structures/src/chain.rs

```diff
@@ -2570,7 +2578,10 @@ impl AltKeys {
     pub fn insert_keys_from_block(&mut self, block: &Block) {
         // Add miner bn256 public keys
         if let Some(value) = block.block_header.bn256_public_key.clone() {
-            self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+            // If the provided key is valid, insert it in altkeys
+            if value.clone().is_valid(){
+                self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+            }
```

Comment:
Maybe we should print a warning in the `else` case. And the same validation needs to be done for commit keys (just below). And add tests :eyes: